willa/web/app.py

```diff
@@ -68,7 +106,9 @@ def _secret() -> str:
     database_url = os.environ.get(
         'DATABASE_URL', f"postgresql://{_pg('USER')}:{_secret()}@{_pg('HOST')}/{_pg('DB')}"
     )
-    return ChainlitDataLayer(database_url=database_url)
+    dl = ChainlitDataLayer(database_url=database_url)
+    dl.get_step = get_step.__get__(dl)
+    return dl
```

Comment:
Horrible monkeypatch until https://github.com/Chainlit/chainlit/pull/2729 is merged.